src/Core/src/Hosting/Dispatching/AppHostBuilderExtensions.cs

```diff
@@ -21,19 +21,40 @@ public static MauiAppBuilder ConfigureDispatching(this MauiAppBuilder builder)
 				if (DispatcherProvider.SetCurrent(provider))
 					svc.CreateLogger<Dispatcher>()?.LogWarning("Replaced an existing DispatcherProvider with one from the service provider.");
 
-				return Dispatcher.GetForCurrentThread()!;
+				var dispatch = Dispatcher.GetForCurrentThread();
+
+				// Slight behavior change 
+				return dispatch ?? svc.GetRequiredService<ApplicationDispatcher>().AppDispatcher;
 			});
-			builder.Services.TryAddEnumerable(ServiceDescriptor.Scoped<IMauiInitializeScopedService, DispatcherInitializer>());
+
+			builder.Services.AddSingleton<ApplicationDispatcher>();
+
+
+			builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IMauiInitializeService, DispatcherInitializer>());
```

Comment:
Is this ok @mattleibow ? 

I can't really tell a difference here between using `IMauiInitializeService` vs `IMauiInitializeScopedService`

The purpose of these are to initialize at the app level not the "Scoped" windows level so it seems like they should just be  `IMauiInitializeService`

Reply:
IMauiInitializeService  should only run once per app during the `MauiAppBuilder.Build()` method. The scoped services are supposed to be initialized for each window. However... I see that it is not working like that so I want to see what is really happening and if this is all a bug because we are not doing something...